fe/src/features/projects/EditProject.vue

```diff
@@ -0,0 +1,183 @@
+                shadow-strength="2"
+                @click="cancelEditProject"
+              />
+              <btn
+                elevated
+                color="negative"
+                label="Delete"
+                type="delete"
+                size="md"
+                class="q-mx-md cornered"
+                glow-color="#00F1AF"
+                shadow-strength="2"
+                @click="deleteProject"
+              />
+              <btn
+                elevated
+                color="positive"
+                label="Update"
+                type="submit"
+                size="md"
+                class="q-mx-md cornered"
+                glow-color="#00F1AF"
+                shadow-strength="2"
+                @click="updateProject"
+              />
+            </div>
+          </div>
+        </div>
+        <div class="row">
+          <BasicInfo />
+          <ProjectCoaches />
+          <ProjectSkills />
+        </div>
+      </div>
+
+      <q-dialog
+        class="full-width"
+        :model-value="deleteProjectID !== -1"
+        persistent
+        @update:model-value="deleteProjectID = -1"
+      >
+        <DeleteProjectDialog
+          :delete-project-id="deleteProjectID"
+          :delete-project-name="deleteProjectName"
+        />
+      </q-dialog>
+    </q-form>
+  </div>
+</template>
+
+<script lang="ts">
+import { ref } from 'vue'
+import { defineComponent, onMounted } from '@vue/runtime-core'
+import { useSkillStore } from '../../stores/useSkillStore'
+import { useCoachStore } from '../../stores/useCoachStore'
+import { useProjectStore } from '../../stores/useProjectStore'
+import router from '../../router'
+import BasicInfo from './components/BasicInfo.vue'
+import ProjectCoaches from './components/ProjectCoaches.vue'
+import ProjectSkills from './components/ProjectSkills.vue'
+import { TempProjectSkill } from '../../models/Skill'
+import DeleteProjectDialog from './components/subcomponents/DeleteProjectDialog.vue'
+
+export default defineComponent({
+  components: { ProjectCoaches, BasicInfo, ProjectSkills, DeleteProjectDialog },
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+  setup(props) {
+    const projectID = ref(props.id)
+    const skillStore = useSkillStore()
+    const coachStore = useCoachStore()
+    const projectStore = useProjectStore()
+
+    const deleteProjectID = ref(-1)
+    const deleteProjectName = ref('')
+
+    onMounted(() => {
+      coachStore.loadUsers()
+      projectStore
+        .getAndSetProject(projectID.value, (skills: TempProjectSkill[]) => {
+          skillStore.setSkills(skills)
+        })
+        // this happens when user requests invalid project id
+        .catch(() => router.push('/projects'))
+    })
+
+    return {
+      projectID,
+      skillStore,
+      coachStore,
+      projectStore,
+      deleteProjectID,
+      deleteProjectName,
+    }
+  },
+  methods: {
+    updateProject() {
```

Comment:
Is it possible to provide some documentation to the functions? Such that we don't need to retroactively add it in the `fe/documenation` branch.